src/course-unit/data/thunk.js

```diff
@@ -202,18 +201,27 @@ export function createNewCourseXBlock(body, callback, blockId, sendMessageToIfra
       });
     } catch (error) {
       dispatch(hideProcessingNotification());
-      dispatch(updateLoadingCourseXblockStatus({ status: RequestStatus.FAILED }));
       handleResponseErrors(error, dispatch, updateSavingStatus);
     }
   };
 }
 
-export function fetchCourseVerticalChildrenData(itemId) {
+export function fetchCourseVerticalChildrenData(itemId, isSplitTestType, skipPageLoading) {
   return async (dispatch) => {
-    dispatch(updateCourseVerticalChildrenLoadingStatus({ status: RequestStatus.IN_PROGRESS }));
+    if (!skipPageLoading) {
+      dispatch(updateCourseVerticalChildrenLoadingStatus({ status: RequestStatus.IN_PROGRESS }));
+    }
 
     try {
       const courseVerticalChildrenData = await getCourseVerticalChildren(itemId);
+      if (isSplitTestType) {
+        const blockIds = courseVerticalChildrenData.children.map(child => child.blockId);
+        const childrenDataArray = await Promise.all(
+          blockIds.map(blockId => getCourseVerticalChildren(blockId)),
+        );
+        const allChildren = childrenDataArray.flatMap(data => data.children || []);
```

Comment:
[optional]: Lets use `reduce` instead of `flatMap`. They have the same algorithmic complexity, but by using `reduce` we can reduce the amount of additional costs due to the lack of intermediate arrays (`map()`). If we have large amounts of data in `childrenDataArray`, the approach with `reduce` should pay for itself.

Reply:
Replaced

Reply:
Resolved